spinalcordtoolbox/spinalcordtoolbox
# Add the canproco nnUNet model for MS lesion segmentation to sct_deepseg

## Description

Opening this PR to add the 2D nnUNet model to segment MS lesions in the spinal cord. The model is a region-based model which means that there are 2 classes (spinal cord and lesion) and the lesions are always located in the spinal cord. The model was trained on STIR and inverted PSIR (multiplied by -1). The trained model can be found in this release: https://github.com/ivadomed/canproco/releases/tag/r20240125. 

I used the 2D model as it is faster than the 3D model and showed similar performances.

## Testing the model

To test this PR, you will need to install SCT from this branch. Here are the steps:

1. Open a terminal, then run `git clone https://github.com/spinalcordtoolbox/spinalcordtoolbox.git` to get a copy of the source code.
2. Run `cd spinalcordtoolbox` to enter the SCT source code folder.
3. Run `git checkout plb/4442_add_canproco_nnunet` to switch to this branch.
4. Run `./install_sct -iy` to install SCT (the installation will exist in this same folder). Make sure that you see all green `[OK]` at the end of the installation.
5. Run `sct_deepseg -install-task seg_sc_ms_lesion_stir_psir` to download the model.
6. Run `sct_deepseg -task seg_sc_ms_lesion_stir_psir -i img.nii.gz` to segment an image.

## Linked issues


This resolves issue #4442
